inflate.c

```diff
@@ -1050,6 +1050,18 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
                 if (INFLATE_NEED_CHECKSUM(strm) && state->wrap & 4) {
                     if (out) {
                         inf_chksum(strm, put - out, out);
+                    } else if ((!state->flags && strm->adler == ADLER32_INITIAL_VALUE)
+#ifdef GUNZIP
+                    || (state->flags && state->crc_fold.value == CRC32_INITIAL_VALUE)) {
+#else
+                        ) {
+#endif
+                        /* In the event we reach this block and the checksums are at the initial values, run a checksum
+                         * over the whole stream, as one of two things are true:
+                         *  1.) We never ran any of the bytes through the checksum (testing has found this happens)
+                         *  2.) We have a checksum that does wrap perfectly, in which case, we should end up still
+                         *      correct, anyway */
+                        inf_chksum(strm, put, strm->total_out);
                     }
```

Comment:
```suggestion
                    } else {
                       bool final_chksum = (!state->flags && strm->adler == ADLER32_INITIAL_VALUE);
#ifdef GUNZIP
                       final_chksum |= (state->flags && state->crc_fold.value == CRC32_INITIAL_VALUE));
#endif
                        /* In the event we reach this block and the checksums are at the initial values, run a checksum
                         * over the whole stream, as one of two things are true:
                         *  1.) We never ran any of the bytes through the checksum (testing has found this happens)
                         *  2.) We have a checksum that does wrap perfectly, in which case, we should end up still
                         *      correct, anyway */
                        if (final_chksum) {
                            inf_chksum(strm, put, strm->total_out);
                    }
```

Reply:
It does require C99 to do this but I don't think that's a huge ask.  We probably already require it.